[nnc] Add stride args while running with allocated outputs

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72223

Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

navahgar committed:
[nnc] Add stride args while running with allocated outputs



Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

[ghstack-poisoned]

eellison commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -1791,8 +1791,19 @@ void TensorExprKernel::runWithAllocatedOutputs(Stack& stack) {
+    // add stride args
+    for (auto idx : c10::irange(input_stride_args_.size())) {
+      const auto& input_stride_arg = input_stride_args_[idx];
+      stride_values[idx] =
```
we can just use the index of tensor stride right ?
```
args.emplace_back(&stack_inputs[input_stride_arg.first].toTensor().strides().at(
              input_stride_arg.second);
```

navahgar replied:
This requires a const_cast:
```
args.emplace_back(const_cast<long*>(
          &stack_inputs[input_stride_arg.first].toTensor().strides().at(
              input_stride_arg.second)));
```

eellison reviewed (approved):
lgtm,  one comment

navahgar committed:
Update on "[nnc] Add stride args while running with allocated outputs"


Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

[ghstack-poisoned]

navahgar committed:
Update on "[nnc] Add stride args while running with allocated outputs"


Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

[ghstack-poisoned]

navahgar committed:
Update on "[nnc] Add stride args while running with allocated outputs"


Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

[ghstack-poisoned]

navahgar committed:
Update on "[nnc] Add stride args while running with allocated outputs"


Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

[ghstack-poisoned]

navahgar committed:
Update on "[nnc] Add stride args while running with allocated outputs"


Differential Revision: [D33960592](https://our.internmc.facebook.com/intern/diff/D33960592/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33960592/)!

[ghstack-poisoned]
